src/functionalTest/java/uk/gov/hmcts/reform/userprofileapi/DeleteRolesToExistingUserFuncTest.java

```diff
@@ -110,4 +112,64 @@ public void should_delete_user_profile_with_roles_successfully() throws Exceptio
         assertThat(!resourceForDeleteCheck.getRoles().contains(puiOrgManager));
 
     }
+
+    @Test
+    public void should_throw_412_while_delete_user_profile_with_invalid_roles_passed() throws Exception {
+        UserProfileCreationData data = createUserProfileData();
+        List<String> roles = new ArrayList<>();
+        roles.add(puiUserManager);
+        String email = idamClient.createUser(roles);
+        data.setEmail(email);
+        UserProfileCreationResponse userProfileCreationResponse = createUserProfile(data, HttpStatus.CREATED);
+
+        RoleName roleDelete = new RoleName("pui-org-manager");
+        Set<RoleName> rolesDelete = new HashSet<>();
+        rolesDelete.add(roleDelete);
+
+        UpdateUserProfileData userProfileDataDelete = new UpdateUserProfileData();
+        userProfileDataDelete.setRolesDelete(rolesDelete);
+
+        UserProfileRolesResponse deleteResourceResp =
+                testRequestHandler.sendDelete(
+                        userProfileDataDelete,
+                        HttpStatus.OK,
+                        requestUri + "/" + userProfileCreationResponse.getIdamId(),

```

Comment:
try to reuse the code and avoide duplication of the code in each and every test method

Reply:
done

Reply:
done